build.gradle

```diff
@@ -60,73 +84,79 @@ pmd {
     ruleSetFiles = files("./qa/pmd.xml")
 }
 
+spotbugs {
+    ignoreFailures = true
+}
+
 checkstyle {
     toolVersion = "7.6.1"
     ignoreFailures = true
     configFile = rootProject.file('./qa/checkstyle.xml')
 }
 
-defaultTasks 'bootRun'
-
-group = 'com.icthh.xm.ms.scheduler'
-version = '0.0.1-SNAPSHOT'
-
-description = ''
-
-bootWar {
-   mainClassName = 'com.icthh.xm.ms.scheduler.SchedulerApp'
-}
-
-war {
-
-}
-
-springBoot {
-    mainClassName = 'com.icthh.xm.ms.scheduler.SchedulerApp'
-    buildInfo()
-}
-
+// See https://virgo47.wordpress.com/2018/09/14/classpath-too-long-with-spring-boot-and-gradle/ for details
+// https://github.com/jhipster/generator-jhipster/issues/9713
 if (OperatingSystem.current().isWindows()) {
-    // https://stackoverflow.com/questions/40037487/the-filename-or-extension-is-too-long-error-using-gradle
     task classpathJar(type: Jar) {
-        dependsOn configurations.runtime
-        appendix = 'classpath'
+        inputs.files sourceSets.main.runtimeClasspath
 
+        archiveName = "runboot-classpath.jar"
         doFirst {
             manifest {
-                attributes 'Class-Path': configurations.runtime.files.collect {
-                    it.toURI().toURL().toString().replaceFirst(/file:\/+/, '/').replaceAll(' ', '%20')
-                }.join(' ')
+                def classpath = sourceSets.main.runtimeClasspath.files
+                attributes "Class-Path": classpath.collect {f -> f.toURI().toString()}.join(" ")
             }
         }
     }
 
     bootRun {
-        dependsOn classpathJar
-        doFirst {
-            classpath = files("$buildDir/classes/java/main", "$buildDir/resources/main", classpathJar.archivePath)
-        }
+        classpath = classpathJar.outputs.files
     }
 }
 
+defaultTasks "bootRun"
+
+war {
+
+}
+
+springBoot {
+    mainClassName = "com.icthh.xm.ms.scheduler.SchedulerApp"
+}
+
 test {
-    exclude '**/CucumberTest*'
```

Comment:
this line must be transferred to new version

Reply:
done